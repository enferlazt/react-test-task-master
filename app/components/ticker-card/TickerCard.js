import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams, withRouter } from 'react-router';
import {connect, disconnect} from '../../services';
import TickerChart from './ticker-chart/TickerChart';
import TickerList from './ticker-list/TickerList';
import PropTypes from 'prop-types';

const TickerCard = (props) => {
    const dispatch = useDispatch();
    const { symbol } = useParams();

    useEffect(() => {
        connect(symbol, dispatch);
        return disconnect;
    }, [props.location]);

    return (
        <section className="ticker-card">
            <TickerList/>
            <TickerChart/>
        </section>
    );
};

TickerCard.propTypes = {
    location: PropTypes.object,
};

export default withRouter(TickerCard);
