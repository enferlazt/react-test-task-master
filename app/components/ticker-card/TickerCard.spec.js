import { shallow } from 'enzyme';
import React from 'react';
import '../../../setupTests';
import TickerCard from './TickerCard';

describe("TickerCard component", () => {
    it("render component", () => {
        const component = shallow(<TickerCard/>);
        expect(component).toMatchSnapshot();
    });
});