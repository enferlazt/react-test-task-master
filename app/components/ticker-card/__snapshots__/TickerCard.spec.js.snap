// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerCard component render component 1`] = `ShallowWrapper {}`;
